docs/how-to/troubleshooting.md

```diff
@@ -1,173 +1,267 @@
+### Troubleshooting providing with Kubo
 
-Given that information, look for another goroutine that might be holding the semaphore in question in the rest of the stack dump.
+With this in mind, if no providers are returned, do the following:
 
-There are a few different reasons that goroutines can be hung:
+1. First, determine how long a reprovide run takes:
 
-- `semacquire` means we're waiting to take a lock or semaphore.
-- `select` means that the goroutine is hanging in a select statement, and none of the cases are yielding anything.
-- `chan receive` and `chan send` are waiting for a channel to be received from or sent on, respectively.
-- `IO wait` generally means that we are waiting on a socket to read or write data, although it *can* mean we are waiting on a very slow filesystem.
+   ```shell
+   ipfs stats reprovide
+   ```
 
-If you see any of those tags _without_ a `, X minutes` suffix, that generally means there isn't a problem -- you just caught that goroutine in the middle of a short wait for something. If the wait time is over a few minutes, that either means that goroutine doesn't do much, or something is pretty wrong.
+   The output should look something like this:
 
-If you see a lot of goroutines, consider using [stackparse](https://github.com/whyrusleeping/stackparse) to filter, sort, and summarize them.
+   ```shell
+   TotalReprovides:       4k (4,140)
+   AvgReprovideDuration:  12.870309s
+   LastReprovideDuration: 44m32.507618s
+   LastReprovide:         2025-06-10 09:45:18
+   ```
 
-### Analyzing the CPU Profile
+2. Note the value for `LastReprovideDuration`. If it is close to 48 hours, select one of the following options, keeping in mind that each has tradeoffs:
 
-The go team wrote an [excellent article on profiling go programs](http://blog.golang.org/profiling-go-programs). If you've already gathered the above information, you can skip down to where they start talking about `go tool pprof`. My go-to method of analyzing these is to run the `web` command, which generates an SVG dotgraph and opens it in your browser. This is the quickest way to easily point out where the hot spots in the code are.
+   - **Enable the [Accelerated DHT Client](https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#accelerated-dht-client) in Kubo**. This configuration improves content providing times significantly by maintaining more connections to peers and a larger routing table and batching advertising of provider records. However, this performance boost comes at the cost of increased resource consumption, most notably network connections to other peers, and can lead to degraded network performance in home networks.
 
-### Analyzing vars and memory statistics
+   - **Change the reprovider strategy from `all` to either `pinned` or `roots`.** In both cases, only provider records for explicitly pinned content are advertised. Differences and tradeoffs are noted below:
```

Comment:
nit: good place to push people towards newly added `pinned+mfs`, and not just `pinned`

```suggestion
   - **Change the [Reprovider Strategy](https://github.com/ipfs/kubo/blob/master/docs/config.md#reproviderstrategy) from `all` to either `pinned+mfs` or `roots`.** In both cases, only provider records for explicitly pinned content are advertised. Differences and tradeoffs are noted below:
```